### PR TITLE
'Scan for Kits' option to Kits' QuickPick

### PR DESCRIPTION
**Pull request by Zingam**
# This change addresses item #864

The following changes are proposed:

- Adds an option to re-scan the kits as a special kit item to the QuickPick.
- Moves all special kits to the top of the list in the QuickPick.

# The purpose of this change

To enable/remind the user to easily invoke the command  Rescan for Kits from the Kits QuickPick.

# Other Notes/Information

I had to resubmit this #893 due to a failed merge.

### CODE REVIEW

**Comment by bobbrow** [comment]
I'm going to postpone this one temporarily until multi-root is committed since much of this code has moved around and I don't want it to get lost.

**Comment by bobbrow** [comment]
Multi-root is committed.  Can you merge and resolve conflicts?

**Comment by Zingam** [comment]
I guess this will need more fixes.

File: src/kit.ts [hunk 1]
```diff
@@ -22,6 +22,14 @@ const localize: nls.LocalizeFunc = nls.loadMessageBundle();
 
 const log = logging.createLogger('kit');
 
+/**
```
Comment (Zingam) [inline src/kit.ts]: @bobbrow I think it makes sense to replace these special kit variables with a enum. Also do we really need to have the type `export type scanforkitsKitType = '__scanforkits__';`

**Comment by Zingam** [comment]
I think this is OK now.

File: src/kit.ts [hunk 2]
```diff
@@ -901,6 +909,9 @@ export async function descriptionForKit(kit: Kit): Promise<string> {
     const compilers = Object.keys(kit.compilers).map(k => `${k} = ${kit.compilers![k]}`);
     return localize('using.compilers', 'Using compilers: {0}', compilers.join(', '));
   }
+  if (kit.name === scanforkitsKitName) {
+    return localize('rescanning.for.kits', 'Rescanning for kits');
```
Comment (bobbrow) [inline src/kit.ts]: "Search for compilers on this computer"

File: src/kit.ts (hunk 1 above)
Comment (bobbrow) [reply src/kit.ts]: I don't have a strong opinion here.

Comment (Zingam) [reply src/kit.ts]: Well I did convert these to an enum in my latest commit.

**Pull request merged**